server/src/main/java/com/cloud/storage/listener/StoragePoolMonitor.java

```diff
@@ -108,10 +108,12 @@ public void processConnect(Host host, StartupCommand cmd, boolean forRebalance)
                 List<StoragePoolVO> zoneStoragePoolsByAnyHypervisor = _poolDao.findZoneWideStoragePoolsByHypervisor(host.getDataCenterId(), HypervisorType.Any);
                 pools.addAll(zoneStoragePoolsByAnyHypervisor);
 
+                // get the disabled pools list if global setting is true.
+                if (StorageManager.MountDisabledStoragePool.value()) {
+                    pools.addAll(_poolDao.findDisabledPoolsByScope(host.getDataCenterId(), null, null, ScopeType.ZONE));
+                }
+
                 for (StoragePoolVO pool : pools) {
-                    if (pool.getStatus() != StoragePoolStatus.Up) {
```

Comment:
Maybe can revert this and extend the condition to `if (pool.getStatus() != StoragePoolStatus.Up && pool.getStatus() != StoragePoolStatus.Disabled)` ?

Reply:
all of the above functions are only returning pools in Up state. so this check is not needed

Reply:
Actually not only Up state since you have added the Disabled pools as well, actually my suggestion above is not correct, should be:

````
if (pool.getStatus() != StoragePoolStatus.Up || (StorageManager.MountDisabledStoragePool.value() && pool.getStatus() != StoragePoolStatus.Disabled && pool.getStatus() != StoragePoolStatus.Up)) {
    continue;
}
````

Reply:
@nvazquez 
this might not be needed, because `findZoneWideStoragePoolsByHypervisor` has the condition `sc.and(sc.entity().getStatus(), Op.EQ, Status.Up);`